post-scan-actions/aws-python-publish-malware-to-sns/template.yml

```diff
@@ -0,0 +1,90 @@
+AWSTemplateFormatVersion: 2010-09-09
+Transform: AWS::Serverless-2016-10-31
+
+Metadata:
+  AWS::ServerlessRepo::Application:
+    Name: cloudone-filestorage-plugin-publish-malware-to-sns
+    Description: >-
+      According to the scan result from the scanner, publish the malware information
+      to the target SNS topic.
+    Author: Trend Micro Cloud One File Storage Security
+
+Parameters:
+  TargetSnsTopicName:
+    Type: String
+    Description: The name for the SNS topic to which malware information publish.
+  PublishMalwareLambdaName:
+    Type: String
+    Description: The name for the lambda function publishes the malware information to SNS topic.
+  PublishMalwarePolicyName:
+    Type: String
+    Description: The name for the policy includes sns and account authority.
+  PublishMalwareRoleName:
+    Type: String
+    Description: The name for the role to execute the lambda function.
+  ScanResultTopicARN:
+    Type: String
+    Description: The ARN of the scan result SNS topic in storage stack.
+Resources:
+  TargetSnsTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: !Ref TargetSnsTopicName
+      DisplayName: Malware notification
+  PublishMalwarePolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      ManagedPolicyName: !Ref PublishMalwarePolicyName
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'sns:TagResource'
+              - 'sns:Publish'
+              - 'account:GetAlternateContact'
+            Resource: '*'
+  PublishMalwareRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Ref PublishMalwareRoleName
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      ManagedPolicyArns:
+        - !Ref PublishMalwarePolicy
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+  PublishMalwareLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      CodeUri: ./
+      Handler: handler.lambda_handler
+      FunctionName: !Ref PublishMalwareLambdaName
+      Runtime: python3.8
+      Timeout: 500
+      Role: !GetAtt
+        - PublishMalwareRole
+        - Arn
+      Environment:
+        Variables:
+          TARGET_SNS_ARN: !Ref TargetSnsTopic
+      MemorySize: 128
+  TopicSubscription:
+    Type: AWS::SNS::Subscription
+    Properties:
+      Protocol: lambda
+      Endpoint: !GetAtt PublishMalwareLambda.Arn
+      TopicArn: !Ref ScanResultTopicARN
+  InvokedLambdaPermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      FunctionName: !GetAtt PublishMalwareLambda.Arn
+      Action: lambda:InvokeFunction
+      Principal: events.amazonaws.com
```

Comment:
This is the endpoint of AWS eventbridge. Should be `sns.amazonaws.com`?

Reply:
Thanks, I'm very confused for why I can't receive the notification from SNS.